### PR TITLE
Switched User Authen to Use HTTP Sessions

### PR DESCRIPTION
**Pull request by christinachance**

### CODE REVIEW

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 1]
```diff
@@ -64,30 +63,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+                    session.setAttribute("name", (String) entity.getProperty("name"));  
+                    session.setAttribute("email", ((String) entity.getProperty("email")).toLowerCase());  
+                    session.setAttribute("password", (String) entity.getProperty("password"));  
+                    session.setAttribute("birthdate", (String) entity.getProperty("birthdate"));  
+                    session.setAttribute("studentId", (long) entity.getProperty("studentId"));  
+                    session.setAttribute("sex", (String) entity.getProperty("sex"));
+                    session.setAttribute("school", (String) entity.getProperty("school"));
+                    session.setAttribute("phone", (String) entity.getProperty("phone"));
+                    session.setAttribute("metric", (String) entity.getProperty("metric")); 
+                    session.setAttribute("admin", (boolean) entity.getProperty("admin"));
```
Comment (rivera-fernando) [inline project/src/main/java/com/google/sps/servlets/LogInServlet.java]: Change back to "is_admin"

Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/LogInServlet.java]: done.

File: project/src/main/java/com/google/sps/servlets/EditUserServlet.java [hunk 2]
```diff
@@ -8,28 +8,24 @@
...
+import com.google.gson.Gson;
+
+
 
 /*gets the user's userId and edits their entity in the datastore and then redirects them to login again*/
 @WebServlet("/edit-user")
 public class EditUserServlet extends HttpServlet {
 
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        HttpSession session=request.getSession(false);
```
Comment (rivera-fernando) [inline project/src/main/java/com/google/sps/servlets/EditUserServlet.java]: spacing probably, just `session = request...`

Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/EditUserServlet.java]: done.

**Review by rivera-fernando — APPROVED** [review]

**Review by Tcintra — APPROVED** [review]

File: project/src/main/java/com/google/sps/servlets/BlobstoreServlet.java [hunk 3]
```diff
@@ -39,9 +39,16 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
     String uploadUrlPost = blobstoreService.createUploadUrl("/post-event");
     String uploadUrlEdit = blobstoreService.createUploadUrl("/edit-event");
+    String uploadUrlEditUser = blobstoreService.createUploadUrl("/edit-user");
+    String uploadUrlSignUp = blobstoreService.createUploadUrl("/user");
+
+
+
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/BlobstoreServlet.java]: remove redundant empty lines.

File: project/src/main/java/com/google/sps/servlets/EditUserServlet.java [hunk 4]
```diff
@@ -46,12 +53,13 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
                     (String)entity.getProperty("password"),
                     (String)entity.getProperty("birthdate"),
-                    Long.parseLong(String.valueOf(entity.getProperty("studentId"))),
+                    (long) entity.getProperty("studentId"),
                     (String)entity.getProperty("sex"),
                     (String)entity.getProperty("school"),
                     (String)entity.getProperty("phone"),
                     (String)entity.getProperty("metric"),
-                    Boolean.parseBoolean(String.valueOf(entity.getProperty("admin"))));
+                    (boolean) entity.getProperty("admin"),
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/EditUserServlet.java:60]: just FYI, usually, we use `is_xx` if the variable is boolean type.

File: project/src/main/java/com/google/sps/servlets/EditUserServlet.java [hunk 5]
```diff
@@ -95,7 +116,51 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+            (long) session.getAttribute("studentId"),
+            (String) session.getAttribute("sex"),
+            (String) session.getAttribute("school"),
+            (String) session.getAttribute("phone"),
+            (String) session.getAttribute("metric"),
+            (boolean) session.getAttribute("admin"),
+            (String) session.getAttribute("imageKey"));
+
+        response.getWriter().println(gson.toJson(user));
+
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/EditUserServlet.java]: remove empty lines

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java [hunk 6]
```diff
@@ -101,9 +89,9 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
         Gson gson = new Gson();
         HttpSession session = request.getSession(false);
-        boolean found = false;
+        Boolean found = false;
         if (session.getAttribute("name") != null) {
-          found = true;
+            found = true;
         }
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/LogInServlet.java]: remove space

File: project/src/main/java/com/google/sps/servlets/RequestAdminServlet.java [hunk 7]
```diff
@@ -25,6 +25,8 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
```
Comment (wulian1223) [inline project/src/main/java/com/google/sps/servlets/RequestAdminServlet.java]: are you using those two in this file?

**Review by wulian1223 — APPROVED** [review]
LGTM after addressing comments.

File: project/src/main/java/com/google/sps/servlets/BlobstoreServlet.java (hunk 3 above)
Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/BlobstoreServlet.java]: done.

File: project/src/main/java/com/google/sps/servlets/EditUserServlet.java (hunk 4 above)
Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/EditUserServlet.java:60]: Thank you for letting me know. I will definitely change that for future pull request.

File: project/src/main/java/com/google/sps/servlets/EditUserServlet.java (hunk 5 above)
Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/EditUserServlet.java]: done.

File: project/src/main/java/com/google/sps/servlets/LogInServlet.java (hunk 6 above)
Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/LogInServlet.java]: done.

File: project/src/main/java/com/google/sps/servlets/RequestAdminServlet.java (hunk 7 above)
Comment (christinachance) [reply project/src/main/java/com/google/sps/servlets/RequestAdminServlet.java]: No, both have been removed.Done.

**Pull request merged**